packages/opentelemetry-basic-tracer/src/Span.ts

```diff
@@ -36,10 +36,12 @@ export class Span implements types.Span {
   private _ended = false;
   private _startTime: number;
   private _endTime = 0;
+  private readonly _logger: Logger;
 
   /** Constructs a new Span instance. */
   constructor(
     parentTracer: types.Tracer,
+    logger: Logger,
```

Comment:
Would it make sense to put this at the end and give it a default value, e.g. the NoopLogger? Or do you think it needs to be required with no default?

Reply:
I'm not a fan of hiding logs by default. Maybe it could be some kind of `ConsoleLogger` by default? Then if the user prefers to use their own logger they can change it. The main issue with this approach is that `console` doesn't have a log level, but that could be configured on the `ConsoleLogger` with a default of `ERROR` or `WARN`.

Reply:
I believe that `console` itself does implement our `Logger` interface, and it has an implicit level of `DEBUG` since it will just print all messages. 

So what if we had something like this in some utility file:

`export const WARN_LEVEL_CONSOLE_LOGGER = console;`

I'm trying to avoid extra layers of wrapping to keep the code minimal if we can, though I'm open to it.

Reply:
> I'm trying to avoid extra layers of wrapping to keep the code minimal if we can, though I'm open to it.

I think the only problem with this is you can't configure the log level. I am in favor of adding `ConsoleLogger` (may be same as https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-core/src/common/console-logger.ts#L25). Let me know I can it in same PR.

Reply:
> I'm trying to avoid extra layers of wrapping to keep the code minimal if we can

The wrapper would not do much except add the concept of a log level to `console`.

Reply:
This should probably be discussed in a new issue to unblock this PR. Let's leave it required for now.

Reply:
Created issue to track https://github.com/open-telemetry/opentelemetry-js/issues/197